flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskExecutor.java

```diff
@@ -2650,11 +2711,16 @@ public void notifyTargetUnreachable(ResourceID resourceID) {
 
         private void handleJobManagerConnectionLoss(ResourceID resourceID, Exception cause) {
             validateRunsInMainThread();
-            jobTable.getConnection(resourceID)
-                    .ifPresent(
-                            jobManagerConnection ->
-                                    disconnectAndTryReconnectToJobManager(
-                                            jobManagerConnection, cause));
```

Comment:
Maybe just pass in `shouldRetainPartitionsOnJobManagerConnectionLost()` as a boolean param. See my other comments.